src/components/PeopleTable.tsx

```diff
@@ -3,3 +1,64 @@
+import { Person } from '../types/Person';
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import { useEffect, useState, useCallback } from 'react';
+import { SearchLink } from '../components/SearchLink';
+import cn from 'classnames';
+
+interface PeopleTableProps {
+  persons: Person[];
+  sortField: string | null;
+  sortOrder: 'asc' | 'desc' | null;
+}
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({
+  persons,
+  sortField,
+  sortOrder,
+}) => {
+  const findPersonByName = (name: string | null) => {
+    return persons.find(p => p.name === name);
+  };
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchParams] = useSearchParams();
+  const { slug } = useParams();
+
+  useEffect(() => {
+    const storedSlug = searchParams.get('selectedPerson');
+
+    if (storedSlug) {
+      const selectPerson = persons.find(person => person.slug === storedSlug);
```

Comment:
Ensure that each person object has a unique 'slug' property. If 'slug' is not guaranteed to be unique, consider using another unique identifier for the key.